### PR TITLE
CMTS-99: Support default values for DeclarationServices

### PR DESCRIPTION
**Pull request by gsfernandes**

### CODE REVIEW

**Comment by gsfernandes** [comment]
--validate

File: modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java [hunk 1]
```diff
@@ -43,20 +50,34 @@
  */
 public class MetadataKeyDeclarationResolver {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetadataKeyDeclarationResolver.class);
```
Comment (lucianoRM) [inline modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java:53]: static import

File: modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java [hunk 2]
```diff
@@ -68,15 +89,31 @@ public MetadataKeyResult resolveKeyResult() {
...
     for (MetadataKeyInfo keyInfo : keyPartModelsInfo) {
-      String id;
+      String id = null;
+      boolean isNullValue = false;
       if (keyPartValues.containsKey(keyInfo.parameterModel.getName())) {
         id = keyPartValues.get(keyInfo.parameterModel.getName());
+        isNullValue = id == null;
+      } else if (!partialKey && keyInfo.parameterModel.getDefaultValue() != null) {
+        String defaultValue = keyInfo.parameterModel.getDefaultValue().toString();
+        if (expressionManager != null && isExpression(defaultValue)) {
```
Comment (lucianoRM) [inline modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java:99]: why this validation? When is the expressionManager going to be null? and why shouldn't we fail if that is the case?

**Review by lucianoRM — CHANGES_REQUESTED** [review]

File: modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java (hunk 2 above)
Comment (gsfernandes) [reply modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/metadata/MetadataKeyDeclarationResolver.java:99]: We use this resolver from MTS to validate that the key is "complete" before calling Runtime. So, on that scenario we won't need to resolve the expression.

**Review by lucianoRM — APPROVED** [review]

**Pull request merged**